### PR TITLE
Migrate.install Hadoop200SE

### PR DESCRIPTION
**Pull request by edquist**
https://twiki.opensciencegrid.org/bin/view/Documentation/Release3/InstallHadoop200SE

- [ ] Enter date into "Migrated" column of google sheet
- [ ] Add migration header to TWiki document

---

This doc still needs tweaking - some remaining INCLUDEs and TWISTYs...

### CODE REVIEW

**Comment by edquist** [comment]
This one is ready for review.

File: docs/data/install-hadoop-2-0-0.md [hunk 1]
```diff
@@ -0,0 +1,1499 @@
+**Hadoop 2.0.0 (CDH4)**
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: We don't need to bold headings

File: docs/data/install-hadoop-2-0-0.md [hunk 2]
```diff
@@ -0,0 +1,1499 @@
...
+=======================
+
+<span class="twiki-macro DOC_STATUS_TABLE"></span>
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Remove TWiki cruft

File: docs/data/install-hadoop-2-0-0.md [hunk 3]
```diff
@@ -0,0 +1,1499 @@
+**Hadoop 2.0.0 (CDH4)**
+=======================
+
+<span class="twiki-macro DOC_STATUS_TABLE"></span> 
+**Purpose**: The purpose of this document is to provide Hadoop based SE administrators the information on how to prepare, install and validate the SE.
+
+!!! warning
+    If you are installing Hadoop/Bestman from OSG 3.1, you will need to use [this](https://twiki.opensciencegrid.org/bin/view/Documentation/Release3/InstallHadoopSE) guide instead. This guide details installing Hadoop 2.0 from the OSG 3.2 repositories.
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Old, delete this warning

File: docs/data/install-hadoop-2-0-0.md [hunk 4]
```diff
@@ -0,0 +1,1499 @@
...
+
+Preparation
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Useless heading, delete

File: docs/data/install-hadoop-2-0-0.md [hunk 5]
```diff
@@ -0,0 +1,1499 @@
...
+[Hadoop Distributed File System](http://hadoop.apache.org/hdfs/) (HDFS) is a scalable reliable distributed file system developed in the Apache project. It is based on map-reduce framework and design of the Google file system. The VDT distribution of Hadoop includes all components needed to operate a multi-terabyte storage site. Included are:
+
+-   An [SRM interface](https://sdm.lbl.gov/srm-wg/doc/SRM.v2.2.html) for grid access;
+-   GridFTP-HDFS as transport layer; and
+-   A [FUSE interface](http://fuse.sourceforge.net/) for localized POSIX access.
+-   [Apache Hadoop](http://hadoop.apache.org/)
+
+The OSG packaging and distribution of Hadoop is based on YUM. All components are packaged as RPMs and are available from the OSG repositories. It is also recommended that you enable [EPEL](http://fedoraproject.org/wiki/EPEL) repos.
+
+Note on upgrading from Hadoop 0.20
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Remove this whole section

> no site is on 0.20 or 0.19

\- BrianB

File: docs/data/install-hadoop-2-0-0.md [hunk 6]
```diff
@@ -0,0 +1,1499 @@
...
+-   **GridFTP node**: This is a node with [Globus GridFTP](http://dev.globus.org/wiki/GridFTP). The GridFTP server for Hadoop can be very memory-hungry, up to 500MB/transfer in the default configuration. You should plan accordingly to provision enough GridFTP servers to handle the bandwidth that your site can support.
+-    **SRM node**: This node will contain the BeStMan SRM frontend for accessing the Hadoop cluster via the SRM protocol. [BeStMan2 SRM](https://sdm.lbl.gov/bestman/)
+
+Note that these components are not necessarily mutually exclusive. For instance, you may consider having your GridFTP server co-located on the SRM node. Alternatively, you can locate a client (or even a GridFTP node) co-located on each data node. That way, each data node also acts as an access point to the hadoop cluster.
+
+Please read the [planning document](https://twiki.opensciencegrid.org/bin/view/Documentation/HadoopUnderstanding) to understand different components of the system.
+
+!!! note
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Replace both e-mails with just the one: goc@opensciencegrid.org

File: docs/data/install-hadoop-2-0-0.md [hunk 7]
```diff
@@ -0,0 +1,1499 @@
...
+
+The HDFS prerequisites are:
+
+-   Minimum of 1 headnode (the namenode)
+-   At least one node which will hold data, preferably at least 2. Most sites will have 20 to 200 datanodes.
+-   Working Yum and RPM installation on every system.
+-   `fuse` kernel module and `fuse-libs`.
+-   Java RPM. If java isn't already installed we supply the Oracle jdk 1.6.0 rpm and it will come in as a dependency. Oracle jdk is currently the only jdk supported by OSG so we highly recommend you use the version supplied.
+
+**Compatibility Note** Note that versions of OpenAFS less than 1.4.7 and greater than 1.4.1 create nameless groups on Linux; these groups confuse Hadoop and prevent its components from starting up successfully. If you plan to install Hadoop on a Linux OpenAFS client, make sure you're running at least OpenAFS 1.4.7.
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Turn this into ```!!! note```

File: docs/data/install-hadoop-2-0-0.md [hunk 8]
```diff
@@ -0,0 +1,1499 @@
...
+
+You will also need a copy of CA certificates (see below). Note that the `osg-se-hadoop-srm` and `osg-se-hadoop-gridftp` package will automatically install a certificate package but will not necessarily pick the cert package you expect. For instance, certain installs will prefer the `osg-ca-scripts` package to fulfill this requirement, which installs a set of scripts to automatically update the certificates, but does not initialize the CA certs by default (you have to run it first). For this reason, you may want to specifically install the cert package of your choice first, before installing Hadoop.
+
+Networking
+----------
+
+For more details on overall Firewall configuration, please see our [Firewall documentation](https://twiki.opensciencegrid.org/bin/view/Documentation/Release3/FirewallInformation).
+
+**NOTE:** The versions of Hadoop in OSG series 3.1 and 3.2 (ie, Hadoop 0.20 and 2.0.0) do not inter-operate. In order to use Hadoop 2.0.0, *all* nodes in the hadoop system (namenode, secondary namenode, datanodes, srm/gridftp nodes and all client nodes) must update to OSG 3.2 and Hadoop 2.0.0.
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Another ```!!! note```, you get the point

File: docs/data/install-hadoop-2-0-0.md [hunk 9]
```diff
@@ -0,0 +1,1499 @@
...
+
+Initializing Certificate Authority
+==================================
+
+This is needed by GridFTP and SRM nodes, but it is recommended for all nodes in the cluster.
+
+++ Enable and Start `fetch-crl` To enable fetch-crl (fetch Certificate Revocation Lists) services by default on the node:
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Why reproduce the entirety of https://opensciencegrid.github.io/docs/common/ca/#install-fetch-crl when you can just link it?

File: docs/data/install-hadoop-2-0-0.md [hunk 10]
```diff
@@ -0,0 +1,1499 @@
...
+Installation
+============
+
+Installation depends on the node you are installing:
+
+Namenode Installation
+---------------------
+
+``` console
+root# yum install osg-se-hadoop-namenode
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Could you fix these prompts using this table? https://opensciencegrid.github.io/technology/software/markdown-migration/#fixing-root-and-user-prompts

Sorry, that's my bad due to a bad conversion in the docker image.

File: docs/data/install-hadoop-2-0-0.md [hunk 11]
```diff
@@ -0,0 +1,1499 @@
...
+-   **Secondary Namenode**: This is a secondary machine that periodically merges updates to the HDFS file system back into the fsimage. This dramatically improves startup and restart times.
+-   **Datanode**: You will have many datanodes. Each data node stores large blocks of files to be stored on the hadoop cluster.
+-   **Client**: This is a documentation shorthand that refers to any machine with the hadoop client commands and [FUSE](http://fuse.sourceforge.net/) mount. Any machine that needs a FUSE mount to access data in a POSIX-like fashion will need this.
+-   **GridFTP node**: This is a node with [Globus GridFTP](http://dev.globus.org/wiki/GridFTP). The GridFTP server for Hadoop can be very memory-hungry, up to 500MB/transfer in the default configuration. You should plan accordingly to provision enough GridFTP servers to handle the bandwidth that your site can support.
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Link to our gridftp doc instead https://opensciencegrid.github.io/docs/data/gridftp/

File: docs/data/install-hadoop-2-0-0.md [hunk 12]
```diff
@@ -0,0 +1,1499 @@
...
+Run the Gratia probe once by hand to check for functionality:
+
+``` console
+root# /usr/share/gratia/gridftp-transfer/GridftpTransferProbeDriver
+```
+
+Look for any abnormal termination and report it if it is a non-trivial site issue. Look in the log files in `/var/log/gratia/<date>.log` and make sure there are no error messages printed.
+
+BeStMan Configuration
+---------------------
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: The contents of this section (except for the "BeStManHadoop-specific configuration" section and the first line of the "Copy certificates to bestman location " section) can be entirely replaced by a link to https://opensciencegrid.github.io/docs/data/bestman-install

File: docs/data/install-hadoop-2-0-0.md [hunk 13]
```diff
@@ -0,0 +1,1499 @@
...
+| /user/cdf       |        0 |            0 | UNKNOWN      | NO QUOTA  |       6 |            0 | UNKNOWN      | 536,870,911,994 |
+| /user/osg       |        1 |            0 | UNKNOWN      | NO QUOTA  |       3 |            0 | UNKNOWN      |   5,368,709,117 |
+| /user/dweitzel  |       20 |            0 | UNKNOWN      | NO QUOTA  |   2,282 |            0 | UNKNOWN      | NO QUOTA        |
+| /user/gattebury |        5 |            0 | UNKNOWN      | NO QUOTA  |  10,002 |            0 | UNKNOWN      | NO QUOTA        |
+| /user/brian     |       72 |            0 | UNKNOWN      | NO QUOTA  |   2,697 |            0 | UNKNOWN      | NO QUOTA        |
+| /user/usatlas   |        0 |            0 | UNKNOWN      | NO QUOTA  |       0 |            0 | UNKNOWN      | NO QUOTA        |
+| /user/powers    |        1 |            1 | UNKNOWN      | NO QUOTA  |     211 |          211 | UNKNOWN      | NO QUOTA        |
+| /user/ifisk     |        0 |            0 | UNKNOWN      | NO QUOTA  |       1 |            0 | UNKNOWN      | NO QUOTA        |
+| /user/gpn       |      261 |           -5 | UNK
```
Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Delete this bullet

Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: We only need to keep this first bullet and if that's the case, it doesn't need to be a list

Comment (brianhlin) [inline docs/data/install-hadoop-2-0-0.md]: Delete this section

**Pull request merged**